### DISCUSSION TITLE
`disable-search` doesn't take effect until the whole action is done

### DISCUSSION THREAD

**Discussion by Kyle0923** [junegunn/fzf#3696] [Q&A] [answered]
I am using the fzf-rg integration like this
```
RG="rg --column --line-number --no-heading --color=always --smart-case "
INITIAL_QUERY="${*:-}"
: | fzf --ansi --disabled --query "$INITIAL_QUERY" \
    --bind "start:reload:$RG {q}" \
    --bind "change:reload:sleep 0.1; $RG {q} || true" \
    --bind "ctrl-t:transform:[[ ! \$FZF_PROMPT =~ ripgrep ]] &&
      echo \"disable-search+reload($RG {q} || true)+rebind(change)+change-prompt(1. ripgrep> )\" ||   <------ here
      echo \"unbind(change)+change-prompt(2. fzf> )+enable-search\"" \
    --color "hl:-1:underline,hl+:-1:underline:reverse" \
    --prompt '1. ripgrep> ' \
    --header 'CTRL-T: Switch between ripgrep/fzf'
```

Note that the query remains untouched during Ctrl-T. The idea is, I can type in fzf mode, realize I am typing regex, and switch over to rg while preserving the query.

But in the line pointed out above, apparently the `disable-search` won't take effect after the whole line is done. The resulting behaviour is after `reload($RG {q})`, fzf will attempt to search in the rg ouputs with the current query, because the query is a regex, fzf would match nothing and show a blank page.

To reproduce the behaviour:
create a simple text file `echo a1 > text`
run the above script. It starts in rg mode, press `ctrl-T` to switch over to fzf mode
type a regex `a\d`, then press `ctrl-T` again to switch back to rg mode

Observe the empty result. The info line shows `0/1` tells us there is 1 candidate and 0 matching. If we type something to trigger the `change` event, the rg searching will kick back in and behave correctly
![image](https://github.com/junegunn/fzf/assets/29618161/99ffac5b-72c3-49c5-9958-1da0ab388434)

I have tried using `reload-sync`, or add a `sleep` at the reload but none seems to work. The closest I got was to add a `change-query(\$FZF_QUERY.)` at the reload line, which can trigger the change event

Is there anyway to force fzf disable-search before reload?

Thanks

**Comment by junegunn** [comment]
I think this actually is a bug. I'll look into it.

**Comment by junegunn** [comment, marked as answer]
https://github.com/junegunn/fzf/commit/008fb9d258e47815a7c36f462bfb0a7fe08ec44f should fix the issue. Please build and test if it works for you. Also you'll have to escape `{q}` in your transform action so that fzf doesn't expand queries like `\d{6}`.

```diff
8c8
<       echo \"disable-search+reload($RG {q} || true)+rebind(change)+change-prompt(1. ripgrep> )\" ||
---
>       echo \"disable-search+reload($RG \\{q} || true)+rebind(change)+change-prompt(1. ripgrep> )\" ||
```

**Reply by Kyle0923** [reply]
I played with it a bit, and I think it is working. Thanks for the quick fix :)
